src/tracker/boosting_tracker.jl

```diff
@@ -0,0 +1,29 @@
+#-----------------------------------
+# Implementation of boosting tracker
+#-----------------------------------
+
+function init_tracker(tracker::TrackerBoosting, boxROI::BoxROI)
+    int_image = integral_image(boxROI.img)
+
+    # sampling
+    tracker.sampler = CurrentSample(tracker.sampler_overlap, tracker.sampler_search_factor, :positive)
```

Comment:
Why do you feel so? I think I didn't get your question.

Reply:
Won't it be clumsy for the user to supply `samplers` each time he updates. I'm thinking of somethine like this :
```julia
mutable struct TrackerBoosting{I <: Integer, F <: AbstractFloat, B <: BoxROI} <: AbstractTracker
    # initialized
    boxROI::B
    num_of_classifiers::I
    initial_iterations::I
    num_of_features::I
    sampler::CurrentSampler
end

abstract type AbstractTrackerSampler end
mutable struct CurrentSampler{F <: Float64, S <: Symbol} <: AbstractTrackerSampler
    overlap::F
    search_factor::F
    mode::S
    sampling_region::SVector{4, Integer}
    valid_region::SVector{4, Integer}
    tracked_patch::SVector{4, Integer}
    CurrentSampler{F, S}(overlap, search_factor, mode) where{F <: Float64, S <: Symbol} = new(overlap, search_factor, mode)
end
```
And initialize `CurrentSampler` in the `constructor` of `TrackerBoosting`

Reply:
I'm not very sure of it since holding the `sampler` field makes these two structures coupled together. Is `CurrentSampler` the only possible sampler used in Boosting tracking algorithm?

as for `initialize(tracker, samplers, box_roi)`
we can support `initialize(tracker, box_roi)` by
```julia
function initialize(tracker::TrackerBoosting, box:: BoxROI)
    sampler = get_sampler()
    initialize(tracker, sampler, box)
end
```
which provides the same but decouples the code.

Reply:
If you think we're making things too complicated, I guess we can revisit this in the future.